doc/api/esm.md

```diff
@@ -1220,6 +1177,50 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
+#### Overriding loader
```

Comment:
Have you tested this code? Can you add it to the examples in https://github.com/nodejs/loaders-test?

We should probably have an example that needs to use both `resolve` and `load`, so that we demonstrate how the two hooks work together. Ideally it would be as realistic a use case as possible.

Speaking of realistic examples, another “`resolve`-only” use case discussed in the design docs for the loaders API was to rewrite specifiers like `lodash` into URLs like `https://esm.sh/lodash`. Perhaps this might be a better example than this “overriding loader,” as it would presumably be a lot shorter of an example.

Reply:
> Have you tested this code? Can you add it to the examples in https://github.com/nodejs/loaders-test?

I have (in my repo that I link to in the issue), but I can definitely copy it to https://github.com/nodejs/loaders-test.

> Speaking of realistic examples, another “resolve-only” use case discussed in the design docs for the loaders API was to rewrite specifiers like lodash into URLs like https://esm.sh/lodash

The example I gave is _very_ realistic as it is a simplification of import maps, where you can map bare specifiers to specific files. I can definitely use your use-case, but I think in terms of complexity, they're roughly the same.

Reply:
Once this lands (hopefully...😊), I'll not only add this loader to the `loaders-test`, but also modify the `http-loader` and `typescript-loader` to not need resolvers.